src/d/actor/d_a_ship.cpp

```diff
@@ -5,460 +5,4713 @@
+                    m0374 = 0;
+                    m0376 = 0;
+                    m0386 = 0;
+                    onStateFlg(daSFLG_UNK1_e);
+                    if (gravity < 0.0f) {
+                        gravity = 0.0f;
+                        current.pos.y += 500.0f;
+                        speed.y = 0.0f;
+                    }
+                    else {
+                        speed.y -= 1.0f;
+                        if (speed.y < -15.0f) {
+                            speed.y = -15.0f;
+                        }
+                    }
+                }
+            }
+            else {
+                return TRUE;
+            }
+        }
+        if (m0298->getPlaySpeed() < 0.01f) {
+            if (m0350 != 4 && m0350 != 0 && mShipMode != m0350) {
+                setPartOnAnime(m0350);
+            }
+        }
+    }
+    else {
+        changeDemoEndProc();
+    }
+    return TRUE;
 }
 
 /* 00006310-00006440       .text procTalkReady_init__8daShip_cFv */
-void daShip_c::procTalkReady_init() {
-    /* Nonmatching */
+BOOL daShip_c::procTalkReady_init() {
+    int iVar1;
+    short sVar2;
+    J3DAnmTransform* pAnimRes;
+    
+    mProc = &daShip_c::procTalkReady;
+    mAnmTransform = NULL;
+    if (dComIfGp_checkPlayerStatus0(0, daPyStts0_SHIP_RIDE_e)) {
+        if ((short)(shape_angle.y - m038C) > 0) {
+            m03B4 = 8;
+        }
+        else {
+            m03B4 = 7;
+        }
+    }
+    else if ((short)(fopAcM_searchPlayerAngleY(this) - shape_angle.y) > 0) {
+        m03B4 = 7;
+    }
+    else {
+        m03B4 = 8;
+    }
+    pAnimRes = (J3DAnmTransform*)dComIfG_getObjectRes(l_arcName, m03B4);
+    m029C->setAnm(pAnimRes, 0, 5.0f, 1.0f, 0.0f, -1.0f, NULL);
+    seStart(JA_SE_SHIP_LOOK_BACK, &eyePos);
+    m034C = 8;
+    return TRUE;
 }
 
 /* 00006440-000064A8       .text procTalkReady__8daShip_cFv */
-void daShip_c::procTalkReady() {
-    /* Nonmatching */
+BOOL daShip_c::procTalkReady() {
+    firstDecrementShipSpeed(0.0f);
+    setMoveAngle(0);
+    if (m029C->getPlaySpeed() <= 0.1f) {
+        procTalk_init();
+    }
+    return TRUE;
 }
 
 /* 000064A8-00006558       .text procTalk_init__8daShip_cFv */
-void daShip_c::procTalk_init() {
-    /* Nonmatching */
+BOOL daShip_c::procTalk_init() {
+    mProc = &daShip_c::procTalk;
+    m034C = 8;
+    m0430 = -1;
+    m038A = 12;
+    mAnmTransform = (J3DAnmTransform*)dComIfG_getObjectRes(l_arcName, m038A);
+    mFrameCtrl.init(mAnmTransform->getFrameMax());
+    setInitMessage();
+    onStateFlg(daSFLG_UNK8000_e);
+    return TRUE;
 }
 
 /* 00006558-00006720       .text procTalk__8daShip_cFv */
-void daShip_c::procTalk() {
-    /* Nonmatching */
+BOOL daShip_c::procTalk() {
+    msg_class *msg;
+    u16 sVar4;
+    
+    firstDecrementShipSpeed(0.0f);
+    setMoveAngle(0);
+    if (m0430 == 0xFFFFFFFF) {
+        if (dComIfGp_checkCameraAttentionStatus(dComIfGp_getPlayerCameraID(0), 4)) {
+            m0430 = fopMsgM_messageSet(mNextMessageNo, &eyePos);
```

Comment:
fopMsgM_messageSet returns a fpc_ProcID, so m0430 can probably be changed to that type, and then also use the invalid ID enum:
```suggestion
    if (m0430 == fpcM_ERROR_PROCESS_ID_e) {
        if (dComIfGp_checkCameraAttentionStatus(dComIfGp_getPlayerCameraID(0), 4)) {
            m0430 = fopMsgM_messageSet(mNextMessageNo, &eyePos);
```